packages/react-reconciler/src/ReactFiberThenable.js

```diff
@@ -212,19 +212,19 @@ export function trackUsedThenable<T>(
             }
           },
         );
+      }
 
-        // Check one more time in case the thenable resolved synchronously.
-        switch (thenable.status) {
-          case 'fulfilled': {
-            const fulfilledThenable: FulfilledThenable<T> = (thenable: any);
-            return fulfilledThenable.value;
-          }
-          case 'rejected': {
-            const rejectedThenable: RejectedThenable<T> = (thenable: any);
-            const rejectedError = rejectedThenable.reason;
-            checkIfUseWrappedInAsyncCatch(rejectedError);
-            throw rejectedError;
-          }
+      // Check one more time in case the thenable resolved synchronously.
+      switch (thenable.status) {
```

Comment:
Not that it's an issue per se, but why did this move into the outer block? Because the `.then(noop)` triggers a lazy initializer?

Reply:
Yes. https://github.com/facebook/react/pull/28284#discussion_r1483763696